hollow/src/main/java/com/netflix/hollow/api/producer/ProducerListenerSupport.java

```diff
@@ -68,81 +67,45 @@ static boolean isValidListener(HollowProducerEventListener l) {
         return LISTENERS.stream().anyMatch(c -> c.isInstance(l));
     }
 
-    private final CopyOnWriteArrayList<HollowProducerEventListener> eventListeners;
-
-    ListenerSupport() {
-        eventListeners = new CopyOnWriteArrayList<>();
+    ProducerListenerSupport() {
+        super();
 
         // @@@ This is used only by HollowIncrementalProducer, and should be
         // separated out
         incrementalCycleListeners = new CopyOnWriteArraySet<>();
     }
 
-    ListenerSupport(List<? extends HollowProducerEventListener> listeners) {
-        eventListeners = new CopyOnWriteArrayList<>(listeners);
+    ProducerListenerSupport(List<? extends HollowProducerEventListener> listeners) {
+        super(listeners);
 
         // @@@ This is used only by HollowIncrementalProducer, and should be
         // separated out
         incrementalCycleListeners = new CopyOnWriteArraySet<>();
     }
 
-    ListenerSupport(ListenerSupport that) {
-        eventListeners = new CopyOnWriteArrayList<>(that.eventListeners);
+    ProducerListenerSupport(ProducerListenerSupport that) {
+        super();
```

Comment:
did you mean to pass in `that` to the `super` call here?

Reply:
Yes, thanks for `that`